internal/wrappers/client.go

```diff
@@ -741,3 +743,12 @@ func extractFromTokenClaims(accessToken, claim string) (string, error) {
 	}
 	return value, nil
 }
+
+func AppendIfNotExists(domains []string, newDomain string) []string {
+	for _, domain := range domains {
```

Comment:
you can change the domains to a map then just check if it's in the map without forloop